pkg/plugin/driver/utils/utils.go

```diff
@@ -63,6 +68,20 @@ func IsPodFailing(pod *v1.Pod) (bool, string) {
 				return true, errstr
 			}
 		}
+
+		// Container terminated without reporting results.
+		if cstatus.State.Terminated != nil {
```

Comment:
No, so this is running in a go routine originated by `go p.Monitor(aChannel)` after starting the plugin `p`.

So the idea is that the plugin should get results and stop monitoring.

However, you're question did make me realize a flow in halting logic: it currently only halts if it is "cleaned up" but (1) I want to make cleanup logic optional (there is a issue for something like this already I think) and (2) the server is only cleaning up the plugins after they _all_ finish.

So if the plugins finish at very different times, pluginA (who finished and reported results) will keep monitoring and then hit this logic eventually.

So what I need to do is either modify the `stop monitoring` flow or add some extra information to this method so that we can check if results are reported yet.

Good catch! 💯

Reply:
So I continued to look at this because something seemed like it was eluding me and now I remember why I considered this safe at the time of writing.

The reason is that _this_ method will report the pod failing, but the action the monitor method takes is not to blow up, but to respond as if there were an error. However, the existing logic will throw out that result if there are already results.

However, that is still not ideal because this will _keep happening_ as long as monitoring is running, so for the reasons above we need to shut it off. So it wont break things, but it would not be a desirable flow and would fill the logs with confusing 'duplicate result' messages.